grcs/0020-knowledge-graph.md

```diff
@@ -199,95 +211,79 @@ Every entity has an implicit Types relation and can have zero, one or many types
-| --------------- |------|----------------------- |
-| Attribute       | Type | GscJ2GELQjmLoaVrYyR3xm |
+### 5. Native value types
 
-The Attribute type has the following attributes:
+There are 6 built in native value types defined. For each native value type, if a value is provided that is invalid, the triple should be dropped and treated as unset. Null cannot be provided as a value. Each native value type is UTF-8 encoded as a string.
 
-| Name           | Type   | Description                                                          |
-| -------------- |--------| -------------------------------------------------------------------- |
-| Value Type     | Text   | The type ID a triple for this attribute should have. There are entities for the native types |
+#### 5.1 Text
 
-Attributes are used for native types like Text, Number and Time. The Value Types should reference the entity IDs created for the native types.
+An arbitrary length string of characters. Text may include newline characters or markdown. Clients may choose which attributes to support markdown for. Example: "Radiohead"
 
-| Name        | ID                     |
-| ------------| ---------------------- |
-| Value Type  | WQfdWjboZWFuTseDhG5Cw1 |
+Options: `language`
 
-The native type IDs are:
+A language option can be provided to specify a language for the text. The language must be an entity ID for a Language entity. If no language is specified and the text is linguistic, it is assumed to be in US English.
 
-| Name         | ID                     |
-| -------------| ---------------------- |
-| Text         | LckSTmjBrYAJaFcDs89am5 |
-| Number       | LBdMpTNyycNffsF51t2eSp |
-| Checkbox     | G9NpD4c7GB7nH5YU9Tesgf |
-| URL          | 5xroh3gbWYbWY4oR3nFXzy |
-| Time         | 3mswMrL91GuYTfBq29EuNE |
-| Point        | UZBZNbA7Uhx1f8ebLi1Qj5 |
+#### 5.2 Number
 
-### 8. Relations
+Numbers are encoded as strings. One decimal point may be used. An optional negative sign (-) can be included at the beginning for negative values. No other symbols such as commas may be used. Additional metadata for units and formatting can be included as options.
 
-Relations describe the edges of a graph. Relations are themselves entities that include details about the relationship. For example a Company can have Team Members. Each Team Member relation can have an attribute describing when the person joined the team. This is a model that is commonly called a property graph.
+Examples:
+* "1234.56"
+* "-789.01"
 
-Relations are ordered. They can be reordered independently without conflicts using Fractional Indexing with an arbitrary level of precision between the indices of the items the user wants to move the item between. A fractional index must include randomness to allow items to be inserted between items with minimal chance of collision. This way moving the order of an item doesn't require changing the index of any of the other items.
+Options: `format`, `unit`
```

Comment:
Instead of having `format` or `unit` I was thinking if these should just be different types e.g.

Number (is a number without any format or unit)
Currency (is a number + an optional currency + format)

The benefit would be that for lots of use-cases where you only need a number you don't need to store/check for any format or unit.